pkg/helm/config.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"fmt"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/kube"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/kubernetes-sigs/minibroker/pkg/log"
+)
+
+const (
+	driver = "secret"
+
+	// Empty values for these mean the internal defaults will be used.
+	defaultKubeConfig = ""
+	defaultContext    = ""
+)
+
+// ConfigProvider defines the signature for a function that provides *action.Configuration.
+type ConfigProvider func(namespace string) (*action.Configuration, error)
+
+// NewDefaultConfigProvider creates a new ConfigProvider using the default dependencies.
+func NewDefaultConfigProvider() ConfigProvider {
+	return NewConfigProvider(
+		log.NewKlog(),
+		DefaultConfigInitializerProvider,
+		defaultKubeConfig,
+		defaultContext,
+	)
+}
+
+// NewConfigProvider creates a new ConfigProvider using the explicit dependencies.
+func NewConfigProvider(
+	log log.Verboser,
+	configInitializerProvider ConfigInitializerProvider,
+	kubeConfig string,
+	context string,
+) ConfigProvider {
+	return func(namespace string) (*action.Configuration, error) {
+		restGetter := kube.GetConfig(kubeConfig, context, namespace)
+		debug := func(format string, v ...interface{}) {
+			log.V(4).Log("helm client: %s", fmt.Sprintf(format, v...))
```

Comment:
```suggestion
			log.V(4).Log("helm client: "+format, v...)
```
That should mean we don't end up calling `Sprintf` if verbosity is less than 4.

Reply:
Awesome!